src/Tools/dotnet-monitor/DiagnosticsMonitorCommandHandler.cs

```diff
@@ -69,36 +69,44 @@ internal sealed class DiagnosticsMonitorCommandHandler
         public async Task<int> Start(CancellationToken token, IConsole console, string[] urls, string[] metricUrls, bool metrics, string diagnosticPort, bool noAuth, bool tempApiKey)
         {
             //CONSIDER The console logger uses the standard AddConsole, and therefore disregards IConsole.
-            using IHost host = CreateHostBuilder(console, urls, metricUrls, metrics, diagnosticPort, noAuth, tempApiKey, configOnly: false).Build();
             try
             {
-                await host.StartAsync(token);
+                using IHost host = CreateHostBuilder(console, urls, metricUrls, metrics, diagnosticPort, noAuth, tempApiKey, configOnly: false).Build();
+                try
+                {
+                    await host.StartAsync(token);
 
-                await host.WaitForShutdownAsync(token);
-            }
-            catch (MonitoringException)
-            {
-                // It is the responsibility of throwers to ensure that the exceptions are logged.
-                return -1;
-            }
-            catch (OptionsValidationException ex)
-            {
-                host.Services.GetRequiredService<ILoggerFactory>()
-                    .CreateLogger(typeof(DiagnosticsMonitorCommandHandler))
-                    .OptionsValidationFailure(ex);
-                return -1;
-            }
-            finally
-            {
-                if (host is IAsyncDisposable asyncDisposable)
+                    await host.WaitForShutdownAsync(token);
+                }
+                catch (MonitoringException)
                 {
-                    await asyncDisposable.DisposeAsync();
+                    // It is the responsibility of throwers to ensure that the exceptions are logged.
+                    return -1;
                 }
-                else
+                catch (OptionsValidationException ex)
                 {
-                    host.Dispose();
+                    host.Services.GetRequiredService<ILoggerFactory>()
+                        .CreateLogger(typeof(DiagnosticsMonitorCommandHandler))
+                        .OptionsValidationFailure(ex);
+                    return -1;
                 }
+                finally
+                {
+                    if (host is IAsyncDisposable asyncDisposable)
+                    {
+                        await asyncDisposable.DisposeAsync();
+                    }
+                    else
+                    {
+                        host.Dispose();
+                    }
+                }
+            } catch (FormatException ex)
+            {
+                TextWriter errorWriter = Console.Error;
+                errorWriter.WriteLine(ex.Message + "\n" + ex.InnerException.Message);
```

Comment:
Break this up into two WriteLine calls, one for the message and one for the inner exception message, which lets the framework decide which set of characters to use for newlines.